buildSrc/src/main/java/org/elasticsearch/gradle/tool/Boilerplate.java

```diff
@@ -18,14 +18,33 @@
  */
 package org.elasticsearch.gradle.tool;
 
+import org.gradle.api.Action;
+import org.gradle.api.NamedDomainObjectContainer;
 import org.gradle.api.Project;
 import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.tasks.SourceSetContainer;
 
+import java.util.Optional;
+
 public abstract class Boilerplate {
 
     public static SourceSetContainer getJavaSourceSets(Project project) {
         return project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets();
     }
 
+    public static <T> T findOrCreate(NamedDomainObjectContainer<T> collection, String name) {
```

Comment:
This is very useful 👍 . My preference is to follow the Gradle API convention of calling this `maybeCreate` but I don't have strong feelings.